core/drivers/crypto/hisilicon/hisi_qm.c

```diff
@@ -780,6 +845,7 @@ enum hisi_drv_status hisi_qp_recv_sync(struct hisi_qp *qp, void *msg)
 	while (true) {
 		ret = hisi_qp_recv(qp, msg);
 		if (ret == HISI_QM_DRVCRYPT_NO_ERR) {
+			udelay(QM_SINGLE_WAIT_TIME);
```

Comment:
How about using `timeout_init_us()` and `timeout_elapsed()` as in `qm_wait_mb_ready()`?